ext/vulnsrc/redhat/ovalv2.go

```diff
@@ -0,0 +1,441 @@
+	Definitions      []*OvalV2Definition   `xml:"definitions>definition"`
+}
+
+// type OvalV2Definitions struct {
+// 	Definitions      []OvalV2Definition   `xml:"oval_definitions>definition"`
+// }
+
+type OvalV2Definition struct {
+	Metadata         OvalV2Metadata    `xml:"metadata"`
+	Criteria         OvalV2Criteria    `xml:"criteria"`
+}
+
+type OvalV2Metadata struct {
+	Advisory         OvalV2Advisory    `xml:"advisory"`
+}
+
+type OvalV2Advisory struct {
+    Issued           OvalV2AdvisoryIssued     `xml:"issued"`
+	Updated          OvalV2AdvisoryUpdated    `xml:"updated"`
+	Severity         string                   `xml:"severity"`
+	Cve              OvalV2CveData            `xml:"cve"`
+	AffectedCpeList  OvalV2Cpe                `xml:"affected_cpe_list"`
+}
+
+type OvalV2AdvisoryIssued struct {
+    Date     string    `xml:"date,attr"`
+}
+
+type OvalV2AdvisoryUpdated struct {
+    Date     string    `xml:"date,attr"`
+}
+
+type OvalV2CveData struct {
+    Cvss3    string    `xml:"cvss3,attr"`
+    Cwe      string    `xml:"cwe,attr"`
+	Href     string    `xml:"href,attr"`
+	Public   string    `xml:"public,attr"`
+}
+
+type OvalV2Cpe struct {
+	Cpe      []string   `xml:"cpe"`
+}
+
+type CpeName struct {
+	Part       string
+	Vendor     string
+	Product    string
+	Version    string
+	Update     string
+	Edition    string
+	Language   string
+}
+
+type OvalV2Criteria struct {
+	Criterion    []OvalV2Criterion    `xml:"criterion"`
+	Criteria     []OvalV2Criteria     `xml:"criteria"`
+}
+
+type OvalV2Criterion struct {
+	XMLName  xml.Name  `xml:"criterion"`
+	Comment  string    `xml:"comment,attr"`
+	TestRef  string    `xml:"test_ref,attr"`
+}
+
+type OvalV2DefinitionNamespaces struct {
+	ModuleNamespaces  []string
+	CpeNamespaces     []CpeName
+}
+
+type RpmNvra struct {
+	Name     string
+	Version  string
+	Release  string
+	Arch     string
+}
+
+// parent call to parse entire doc
+func ParseDefinitionNamespaces(ovalDefinitionsXml string) []OvalV2DefinitionNamespaces {
+	ovalV2DefinitionNamespaces := []OvalV2DefinitionNamespaces{}
+	ovalDefinitions := OvalV2Definitions{}
+	err := xml.Unmarshal([]byte(ovalDefinitionsXml), &ovalDefinitions)
+	if err != nil {
+		panic(err)
+	}
+	for _, definition := range ovalDefinitions.Definitions {
+		criteriaNs, err := ParseCriteriaForModuleNamespaces(*definition)
+		if err != nil {
+			// log error and continue
+			log.Error(err)
+		}
+		cpeNames, err := ParseParseCpeNameFromAffectedCpeList(definition.Metadata.Advisory.AffectedCpeList)
+		if err != nil {
+			// log error and continue
+			log.Error(err)
+		}
+		ovalV2DefinitionNamespace := OvalV2DefinitionNamespaces{criteriaNs, cpeNames}
+		ovalV2DefinitionNamespaces = append(ovalV2DefinitionNamespaces, ovalV2DefinitionNamespace)
+	}
+	return ovalV2DefinitionNamespaces
+}
+
+// parse one definition
+func ParseCriteriaForModuleNamespaces(ovalDefinition OvalV2Definition) ([]string, error) {
```

Comment:
The function doesn't return any error object. If you are not going to extend the function please remove it.